File: ace-home/src/main/java/com/acehome/entities/bank/Bank.java
package com.acehome.entities.bank;

import java.util.Set;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import com.acehome.entities.Activity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity()
@Table(name="bank")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class Bank {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column()
    private String name;

    @Column()
    private String number;
    
    @Column()
    private String description;

    @OneToMany(mappedBy="bank")
    private Set<Activity> activities;

    @Column()
    private boolean isDeleted;
}

File: ace-home-cli/src/app/pages/bank/models/bank.ts
export class Bank {
    id!: number;
    name!: string;
    number!: string;
    description!: string;
    devise!:string; 

    constructor() {
        this.id = 0;
        this.name = "name";
        this.number = "number";
        this.description = "description";
        this.devise = "EUR";
    }
}
